### PR TITLE
Add solution for lab02

### PR DESCRIPTION
**Pull request by lucigrigo**

### CODE REVIEW

File: labs/lab02/skel/win/2-crc/crc.c [hunk 1]
```diff
@@ -20,119 +20,95 @@
...
-	/* TODO 1 - Open source file for reading */
+    /* TODO 1 - Open source file for reading */
 
-	/* TODO 1 - Create destination file for writing */
+    /* TODO 1 - Create destination file for writing */
 
-	/* Read from file  */
-	while (1) {
-		ZeroMemory(buf, sizeof(buf));
+    /* read from file  */
```
Comment (teodutu) [inline labs/lab02/skel/win/2-crc/crc.c]: Keep the capital letter for `read` so it's consistent with the rest of the comments.

File: labs/lab02/skel/win/2-crc/crc.c [hunk 2]
```diff
@@ -20,119 +20,95 @@
 #define BUFSIZE 512
 #define CHUNKSIZE 32
 
-static void WriteCrc(int crc, HANDLE hWrite)
-{
-	BOOL bRet;
-	DWORD dwBytesWritten, dwBytesToWrite, dwTotalWritten;
-	/* TODO 1 - Write the CRC to the file. Use a loop! */
-}
```
Comment (teodutu) [inline labs/lab02/skel/win/2-crc/crc.c]: Why delete this? It is necessary to write the CRC to the file.

File: labs/lab02/skel/win/2-crc/crc.c [hunk 3]
```diff
@@ -20,119 +20,95 @@
...
+    }
 
-	/* TODO 2 - Calculate and return file size using SetFilePointer */
+    /* TODO 1 - Write crc to destination file */
 
-	return dwSize;
+    /* TODO 1 - Close files */
 }
 
-static DWORD ReadChunk(CHAR *chunk, HANDLE hFile)
```
Comment (teodutu) [inline labs/lab02/skel/win/2-crc/crc.c:70]: Why remove this function?

File: labs/lab02/skel/win/2-crc/crc.c [hunk 4]
```diff
@@ -20,119 +20,95 @@
...
-		
-		/* TODO 4 - Compare the previously read chunks */
-	}
+    /* TODO 3 - Compare the CRC files, chunk by chunk */
+    /*
+     * while (1) {
+     *	ZeroMemory(chunk1, sizeof(chunk1));
+     *	ZeroMemory(chunk2, sizeof(chunk2));
+     * }
+     */
```
Comment (teodutu) [inline labs/lab02/skel/win/2-crc/crc.c]: I don't think this is a better approach. The current skeleton and TODOs look clearer to me. Yes, they are more handholdy, but this exercise is more difficult than the others, so I think the more granular TODOs are justified.

File: labs/lab02/skel/win/3-ls/ls.c [hunk 5]
```diff
@@ -32,123 +32,134 @@ static void PrintFileTime(FILETIME fTime);
 /*
  * List files from directory reprezented by path
  * @path        - path to dir we want to list
- * @bRecursive  - if "ls -R" was issued
+ * @bRecursive  - if "ls -r" was issued
```
Comment (teodutu) [inline labs/lab02/skel/win/3-ls/ls.c]: Should be `-R` so it's consistent with the POSIX `ls`.

File: labs/lab02/skel/win/3-ls/ls.c [hunk 6]
```diff
@@ -32,123 +32,134 @@ static void PrintFileTime(FILETIME fTime);
...
+            bAll = TRUE;
+            continue;
+        }
+        if (strcmp(argv[i], "-R") == 0)
+        {
+            bRecursive = TRUE;
+            continue;
+        }
+
+        /* save path */
```
Comment (teodutu) [inline labs/lab02/skel/win/3-ls/ls.c]:
```suggestion
        /* Save path */

```

File: labs/lab02/sol/lin/3-mcat/mcat.c [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+	if (argc < 1 || argc > 3) {
+		printf("Usage:\n ./cat source_file [destination_file]\n");
+		return 0;
+	}
+
+	/* TODO 1 - open source file for reading */
+	fd = open(argv[1], O_RDONLY);
+	DIE(fd < 0, "open source");
+
```
Comment (teodutu) [inline labs/lab02/sol/lin/3-mcat/mcat.c:37]:
```suggestion
```

File: labs/lab02/sol/lin/3-mcat/mcat.c [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+	if (argc == 3) {
+		/* TODO 2 - redirect stdout to destination file */
+
```
Comment (teodutu) [inline labs/lab02/sol/lin/3-mcat/mcat.c:40]:
```suggestion

```

File: labs/lab02/sol/win/1-cat/cat.c [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    hFile = CreateFile(argv[1],
+                       GENERIC_READ,
+                       0,
+                       NULL,
+                       OPEN_EXISTING,
+                       FILE_ATTRIBUTE_NORMAL,
+                       NULL);
+    DIE(hFile == INVALID_HANDLE_VALUE, "CreateFile argv[0]");
+
+    /* read file pointed by argv[1] and send it to standadard output */
```
Comment (teodutu) [inline labs/lab02/sol/win/1-cat/cat.c]:
```suggestion
    /* Read file pointed by argv[1] and send it to standadard output */
```

File: labs/lab02/sol/win/1-cat/cat.c [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    BOOL bRet;
+    DWORD bytesRead, bytesWritten, totalWritten;
+
+    if (argc != 2)
+    {
+        fprintf(stderr, "Usage: %s <file>\n", argv[0]);
+        exit(EXIT_FAILURE);
+    }
+
+    /* open file with name pointed by argv[1] */
```
Comment (teodutu) [inline labs/lab02/sol/win/1-cat/cat.c]:
```suggestion
    /* Open file with name pointed by argv[1] */
```

File: labs/lab02/sol/win/1-cat/cat.c [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                        buf,
+                        BUFSIZE,
+                        &bytesRead,
+                        NULL);
+        DIE(bRet == FALSE, "ReadFile");
+
+        if (bytesRead == 0)
+            break;
+
+        /* how much we've actually written */
```
Comment (teodutu) [inline labs/lab02/sol/win/1-cat/cat.c]:
```suggestion
        /* How much we've actually written */
```

File: labs/lab02/sol/win/1-cat/cat.c [hunk 12]
```diff
@@ -0,0 +1,75 @@
+/**
+ * SO
+ * Lab #2, Simple I/O operations
+ *
+ * Task #1, Windows
+ *
+ * Implementing cat
+ */
+
+/* do not use UNICODE */
```
Comment (teodutu) [inline labs/lab02/sol/win/1-cat/cat.c]:
```suggestion
/* Do not use UNICODE */
```

File: labs/lab02/sol/lin/5-singular/singular.c [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+    sleep(10);
+}
+
+static void check_lock(void)
+{
+    int rc;
+
+    /* TODO - open file */
+    /* O alta solutie, pentru a limita apelurile de sistem, este
+     * deschiderea unui fisier cu flag-urile O_EXCL si O_CREAT
```
Comment (teodutu) [inline labs/lab02/sol/lin/5-singular/singular.c]: Translate this to English.

File: labs/lab02/sol/lin/4-trouble/trouble.c [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "utils.h"
+
+int main(void)
+{
+    int fd, rc;
+    size_t bytes_written = 0, count;
+    char *msg = "Ana are mere!\n";
+
+    /* opening files */
```
Comment (teodutu) [inline labs/lab02/sol/lin/4-trouble/trouble.c]:
```suggestion
    /* Open files */
```

File: labs/lab02/sol/win/2-crc/crc.c [hunk 15]
```diff
@@ -0,0 +1,244 @@
+/**
+ * SO
+ * Lab #2, Simple I/O operations
+ *
+ * Task #2, Windows
+ *
+ * Implementing grep
+ */
+
+/* do not use UNICODE */
```
Comment (teodutu) [inline labs/lab02/sol/win/2-crc/crc.c]:
```suggestion
/* Do not use UNICODE */
```

File: labs/lab02/sol/lin/5-singular/singular.c [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+static void do_stuff(void)
+{
+    sleep(10);
+}
+
+static void check_lock(void)
+{
+    int rc;
+
+    /* TODO - open file */
```
Comment (teodutu) [inline labs/lab02/sol/lin/5-singular/singular.c]:
```suggestion
    /* TODO - Open file */
```

File: labs/lab02/sol/lin/5-singular/singular.c [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+    /* O alta solutie, pentru a limita apelurile de sistem, este
+     * deschiderea unui fisier cu flag-urile O_EXCL si O_CREAT
+     */
+    fdlock = open(LOCK_FILE, O_CREAT | O_RDWR, 0644);
+    DIE(fdlock < 0, "open fdlock");
+
+    /* TODO - lock the file using flock
```
Comment (teodutu) [inline labs/lab02/sol/lin/5-singular/singular.c]:
```suggestion
    /* TODO - Lock the file using flock
```

File: labs/lab02/sol/lin/5-singular/singular.c [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        DIE(rc, "flock failed");
+    }
+
+    printf("\nGot Lock\n\n");
+}
+
+static void clean_up(void)
+{
+    /* TODO - unlock file, close file and delete it */
```
Comment (teodutu) [inline labs/lab02/sol/lin/5-singular/singular.c]:
```suggestion
    /* TODO - Unlock file, close file and delete it */
```

File: labs/lab02/sol/win/2-crc/crc.c [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+                        NULL,
+                        CREATE_ALWAYS,
+                        FILE_ATTRIBUTE_NORMAL,
+                        NULL);
+    DIE(hWrite == INVALID_HANDLE_VALUE, "CreateFile");
+
+    /* read from file  */
+    while (1)
+    {
+
```
Comment (teodutu) [inline labs/lab02/sol/win/2-crc/crc.c]:
```suggestion
```

File: labs/lab02/sol/win/2-crc/crc.c [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+    hWrite = CreateFile(destFile,
+                        GENERIC_WRITE,
+                        0,
+                        NULL,
+                        CREATE_ALWAYS,
+                        FILE_ATTRIBUTE_NORMAL,
+                        NULL);
+    DIE(hWrite == INVALID_HANDLE_VALUE, "CreateFile");
+
+    /* read from file  */
```
Comment (teodutu) [inline labs/lab02/sol/win/2-crc/crc.c]:
```suggestion
    /* Read from file  */
```

File: labs/lab02/sol/win/2-crc/crc.c [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+                        BUFSIZE,
+                        &bytesRead,
+                        NULL);
+        DIE(bRet == FALSE, "ReadFile");
+
+        /* TODO 1 - Test for end of file */
+        if (bytesRead == 0)
+            break;
+
+        /* calculate crc for buf */
```
Comment (teodutu) [inline labs/lab02/sol/win/2-crc/crc.c]:
```suggestion
        /* Calculate crc for buf */
```

File: labs/lab02/sol/win/3-ls/ls.c [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+ * Lab #2, Simple I/O operations
+ *
+ * Task #3, Windows
+ *
+ * Implementing "ls" tool with options
+     -a - print attributes of the file
+     -r - recursive
+ */
+
+/* do not use UNICODE */
```
Comment (teodutu) [inline labs/lab02/sol/win/3-ls/ls.c]:
```suggestion
/* Do not use UNICODE */
```

File: labs/lab02/sol/win/3-ls/ls.c [hunk 23]
```diff
@@ -0,0 +1,198 @@
...
+#undef UNICODE
+
+#include <windows.h>
+#include <stdio.h>
+
+#include <utils.h>
+
+#define MAXPATH 256
+
+/* enough to fit something like "1-ls.vcproj.CHOOCH.Administrator.user" */
```
Comment (teodutu) [inline labs/lab02/sol/win/3-ls/ls.c]:
```suggestion
/* Enough to fit something like "1-ls.vcproj.CHOOCH.Administrator.user" */
```

File: labs/lab02/sol/win/3-ls/ls.c [hunk 24]
```diff
@@ -0,0 +1,198 @@
...
+static void PrintFileTime(FILETIME fTime);
+
+/* Print program parameters and usage */
+static void PrintUsage(char *program);
+
+/*
+ * List files from directory reprezented by path
+ * @path		- path to dir we want to list
+ * @bRecursive	- if "ls -r" was issued
+ * @bAll		- if "ls -a" was issued
```
Comment (teodutu) [inline labs/lab02/sol/win/3-ls/ls.c]:
```suggestion
 * List files from directory reprezented by path
 * @path	- path to dir we want to list
 * @bRecursive	- if "ls -R" was issued
 * @bAll	- if "ls -a" was issued
```

File: labs/lab02/sol/win/3-ls/ls.c [hunk 25]
```diff
@@ -0,0 +1,198 @@
+/**
+ * SO
+ * Lab #2, Simple I/O operations
+ *
+ * Task #3, Windows
+ *
+ * Implementing "ls" tool with options
+     -a - print attributes of the file
+     -r - recursive
```
Comment (teodutu) [inline labs/lab02/sol/win/3-ls/ls.c]:
```suggestion
     -R - recursive
```

File: labs/lab02/skel/win/3-ls/ls.c [hunk 26]
```diff
@@ -1,12 +1,12 @@
...
+ * SO
+ * Lab #2, Simple I/O operations
  *
  * Task #3, Windows
  *
  * Implementing "ls" tool with options
-	 -a - print attributes of the file
-	 -r - recursive
+     -a - print attributes of the file
+     -r - recursive
```
Comment (teodutu) [inline labs/lab02/skel/win/3-ls/ls.c]:
```suggestion
     -R - recursive
```

File: labs/lab02/sol/win/4-trouble/trouble.c [hunk 27]
```diff
@@ -0,0 +1,53 @@
+/**
+ * SO
+ * Lab #2, Simple I/O operations
+ *
+ * Task #4, Windows
+ *
+ * Troubleshoot
+ */
+
+/* do not use UNICODE */
```
Comment (teodutu) [inline labs/lab02/sol/win/4-trouble/trouble.c]:
```suggestion
/* Do not use UNICODE */
```

File: labs/lab02/sol/win/utils/utils.h [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+ * Useful structures/macros
+ */
+
+#ifndef WIN_UTILS_H_
+#define WIN_UTILS_H_	1
+
+#include <windows.h>
+#include <stdio.h>
+
+/* useful macro for handling error codes */
```
Comment (teodutu) [inline labs/lab02/sol/win/utils/utils.h]:
```suggestion
/* Useful macro for handling error codes */
```

**Review by teodutu — CHANGES_REQUESTED** [review]
I left some required changes as inline suggestions. The rest of the suggestions apply globally:

- Remove the `.DS_Store` file
- Revert the indentation from 4 spaces to tabs.
- Use the following style for multiline comments:
```
/**
 *
 */
```
- Use the Linux Kernel coding style for multiline `if`s, `for`s etc:
```c
if (cond) {
	/* stuff */
}
```
- In general, run `checkpatch.pl` and fix the coding style. Ignore the lack of an SPDX license and the CRLF line endings in Windows files.
- Why delete `lab02/skel/win/2-crc/tests/tests_out`? They're useful for veryfing if the CRC is calculated correctly.
- Move `lab02/skel/win/2-crc/tests_in` back to ``lab02/skel/win/2-crc/tests/tests_in`. i.e. put them back inside the `tests` directory, together with `tests_out`.
- Add `lab02/sol/win/2-crc/tests_out` and place them in a `tests` folder, together with `tests_in`, just like the tests files for the skeleton.
- Remove `labs/lab02/sol/.gitignore`. It's empty.
- Remove the empty line after the block comments at the beginning of `sol` files.

File: labs/lab02/skel/win/2-crc/crc.c (hunk 2 above)
Comment (lucigrigo) [reply labs/lab02/skel/win/2-crc/crc.c]: this file from skel folder from so-internal did not contain the function, but the sol file did
when I copied the files I forgot to add it back

File: labs/lab02/skel/win/2-crc/crc.c (hunk 3 above)
Comment (lucigrigo) [reply labs/lab02/skel/win/2-crc/crc.c:70]: same as above

**Comment by lucigrigo** [comment]
Apparently I trusted my code formatting script too much (or too blindly :D).
I checked and fixed each file manually, hope I did not forget anything.

File: labs/lab02/skel/win/2-crc/crc.c [hunk 29]
```diff
@@ -1,114 +1,137 @@
...
+		equal = CompareFiles(argv[2], argv[3]);
+
+		if (equal)
+			printf("Files are equal\n");
+		else
+			printf("Files differ\n");
+	}
+
+	return 0;
+}
```
Comment (teodutu) [inline labs/lab02/skel/win/2-crc/crc.c]: Add a newline at the end of the file.

**Review by teodutu — CHANGES_REQUESTED** [review]
It's almost alright. There are a few more things you need to fix:
- Remove all `.DS_Store` files.
- Add the `tests_out` directory and the files inside it to `la02/skel/win/2-crc/tests`.
- Use tabs (`\t`), not spaces to align the `\`s within the `DIE` macros in all `utils.h` files.
- Capitalize the comments in `lab02/skel/lin/nasty/nasty.c`.

File: labs/lab02/skel/win/3-ls/ls.c [hunk 30]
```diff
@@ -46,53 +46,49 @@ static void ListFiles(TCHAR *path, BOOL bRecursive, BOOL bAll)
...
 			 */
 
 			if (bRecursive == TRUE) {
-				/* 
-				 * TODO 3 - Recursive
-				 * Remember that . and .. are also members of the current
-				 * directory. Do not include them in your recursion, or you
-				 * might loop infinitely
-				 */
+				/* TODO 3 - Recursive */
```
Comment (andreia-oca) [inline labs/lab02/skel/win/3-ls/ls.c]: Why did you delete this hint? I think it makes sense to keep it.

File: labs/lab02/skel/win/3-ls/ls.c [hunk 31]
```diff
@@ -46,53 +46,49 @@ static void ListFiles(TCHAR *path, BOOL bRecursive, BOOL bAll)
...
 				 * Use the PrintFileTime function
 				 */
 
-				/* TODO 2 - get the file size and print it */
+				/* TODO 2 - Get File Size */
 			}
 			printf("\n");
 		}
-	} while (/* TODO 1 - interate if there are still files in list */ 0);
+	} while (/* TODO 1 - Interate if there are still files in list */ 0);
```
Comment (andreia-oca) [inline labs/lab02/skel/win/3-ls/ls.c]: Typo `interate` -> `Iterate`

File: labs/lab02/sol/lin/3-mcat/mcat.c [hunk 32]
```diff
@@ -0,0 +1,67 @@
...
+#include <unistd.h>
+
+#include "utils.h"
+#include "xfile.h"
+
+#define BUFSIZE 32
+
+int main(int argc, char **argv)
+{
+	int fd, fd2, rc, bytesRead;
```
Comment (andreia-oca) [inline labs/lab02/sol/lin/3-mcat/mcat.c]:
```suggestion
	int fd; 
	int fd2;
	int rc;
	int bytesRead;
```

File: labs/lab02/sol/lin/4-trouble/trouble.c [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "utils.h"
+
+int main(void)
+{
+	int fd, rc;
```
Comment (andreia-oca) [inline labs/lab02/sol/lin/4-trouble/trouble.c]:
```suggestion
	int fd;
	int rc;
```

File: labs/lab02/sol/win/1-cat/cat.c [hunk 34]
```diff
@@ -0,0 +1,72 @@
...
+#include <utils.h>
+
+#define BUFSIZE 8
+
+int main(int argc, char *argv[])
+{
+	HANDLE hFile;
+	CHAR buf[BUFSIZE];
+	BOOL bRet;
+	DWORD bytesRead, bytesWritten, totalWritten;
```
Comment (andreia-oca) [inline labs/lab02/sol/win/1-cat/cat.c]:
```suggestion
	DWORD bytesRead;
	DWORD bytesWritten;
	DWORD totalWritten;
```

File: labs/lab02/sol/win/2-crc/crc.c [hunk 35]
```diff
@@ -0,0 +1,234 @@
...
+#include <utils.h>
+#include <crc32.h>
+
+#define BUFSIZE 512
+#define CHUNKSIZE 32
+
+static void WriteCrc(int crc, HANDLE hWrite)
+{
+	BOOL bRet;
+	DWORD dwBytesWritten, dwBytesToWrite = sizeof(crc);
```
Comment (andreia-oca) [inline labs/lab02/sol/win/2-crc/crc.c]:
```suggestion
	DWORD dwBytesWritten;
	DWORD dwBytesToWrite = sizeof(crc);
```

File: labs/lab02/sol/win/3-ls/ls.c [hunk 36]
```diff
@@ -0,0 +1,187 @@
...
+ * @bAll		- if "ls -a" was issued
+ */
+static void ListFiles(TCHAR *path, BOOL bRecursive, BOOL bAll)
+{
+	WIN32_FIND_DATA ffd;
+	FILETIME ftWrite;
+	LARGE_INTEGER filesize;
+	HANDLE hFind = INVALID_HANDLE_VALUE;
+	DWORD dwRet;
+	TCHAR extendedPath[MAXPATH], newPath[MAXPATH];
```
Comment (andreia-oca) [inline labs/lab02/sol/win/3-ls/ls.c]:
```suggestion
	TCHAR extendedPath[MAXPATH];
	TCHAR newPath[MAXPATH];

```

**Review by andreia-oca — CHANGES_REQUESTED** [review]
Looks good to me. General suggestions are:
 - One variable declaration per line.
 - Delete the extra newline between the first comments and `include` from each source file.
 - The code should be indented using tabs, not spaces. (At least `/so/labs/lab02/sol/lin/3-mcat/xfile.c` is indented with spaces right now).
 
Use [checkpatch.pl](https://github.com/torvalds/linux/blob/master/scripts/checkpatch.pl) to check the indentation and potential trailing white spaces.

**Comment by lucigrigo** [comment]
I believe that the newline between introduction comments and `/* Do not use UNICODE */` is fine. Where UNICODE comment is missing, that newline does not exist.

Double-checked all source files for space indent, but all of them were previously indented using tabs.

File: labs/lab02/skel/win/2-crc/tests/test_out/test1.out [hunk 37]
```diff
@@ -1 +0,0 @@
-����
```
Comment (teodutu) [inline labs/lab02/skel/win/2-crc/tests/test_out/test1.out]: Add this back. Do the same for the other tests (`test2.out`, `test3.out`) and copy them to the `sol` directory as well.

File: labs/lab02/skel/lin/utils/utils.h [hunk 38]
```diff
@@ -4,22 +4,23 @@
...
-					__FILE__, __LINE__);		\
-			perror(call_description);			\
-			exit(EXIT_FAILURE);				\
-		}							\
+/* Useful macro for handling error codes */
+#define DIE(assertion, call_description)					\
+	do									\
+	{									\
+		if (assertion)							\
+		{								\
```
Comment (teodutu) [inline labs/lab02/skel/lin/utils/utils.h]:
```suggestion
#define DIE(assertion, call_description)					\
	do {									\
		if (assertion) {						\
```

File: labs/lab02/skel/win/utils/utils.h [hunk 39]
```diff
@@ -1,25 +1,27 @@
...
-					__FILE__, __FUNCTION__, __LINE__); \
-			PrintLastError(call_description); \
-			exit(EXIT_FAILURE); \
-		} \
+/* Useful macro for handling error codes */
+#define DIE(assertion, call_description)					\
+	do									\
+	{									\
+		if (assertion)							\
+		{								\
```
Comment (teodutu) [inline labs/lab02/skel/win/utils/utils.h]:
```suggestion
#define DIE(assertion, call_description)					\
	do {									\
		if (assertion) {						\
```

File: labs/lab02/sol/win/utils/utils.h [hunk 40]
```diff
@@ -0,0 +1,42 @@
...
+
+#include <windows.h>
+#include <stdio.h>
+
+/* Useful macro for handling error codes */
+#define DIE(assertion, call_description)					\
+	do									\
+	{									\
+		if (assertion)							\
+		{								\
```
Comment (teodutu) [inline labs/lab02/sol/win/utils/utils.h]:
```suggestion
#define DIE(assertion, call_description)					\
	do {									\
		if (assertion) {						\
```

**Review by teodutu — CHANGES_REQUESTED** [review]
Just place the brackets on the same line as the `do` and `if` keywords and it's fine.

**Review by teodutu — APPROVED** [review]
Looks good!

**Review by andreia-oca — APPROVED** [review]
Looks good to me

**Pull request merged**